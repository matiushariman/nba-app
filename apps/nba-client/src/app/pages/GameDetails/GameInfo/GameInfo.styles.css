@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .hero-layer {
    @apply absolute object-cover w-full h-full top-0 right-0 bottom-0 left-0;
  }
}

.hero-overlay {
  background: linear-gradient(90deg,#000 -3.8%,transparent 52%),linear-gradient(180deg,transparent,#000 78.7%);
  z-index: 1;
}
