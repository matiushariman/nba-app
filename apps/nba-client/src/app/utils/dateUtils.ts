import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
