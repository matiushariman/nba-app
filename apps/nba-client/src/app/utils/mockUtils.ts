export const shouldUseMock = process.env['NODE_ENV'] === 'development';
