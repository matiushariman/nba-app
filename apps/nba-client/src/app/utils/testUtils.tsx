import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
