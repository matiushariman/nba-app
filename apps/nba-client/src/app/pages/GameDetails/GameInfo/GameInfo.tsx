import TeamLogo from '../../../components/TeamLogo';
import { selectGame } from '../../../hooks/useGetGame';
import { selectArena, selectGameTime } from '../../../hooks/useGetBoxscore';
import { convertDateToEt } from '../../../utils/dateUtils';

import './GameInfo.styles.css';

const GameInfo = () => {
  const game = selectGame();
  const arena = selectArena();
  const gameTime = selectGameTime();

  if (!game || !arena || !gameTime) {
    return null;
  }

  const { awayTeam, homeTeam } = game;

  return (
    <div className="bg-black flex">
      <div className="flex-grow opacity-20 min-w-[240px] relative grayscale hidden md:block">
        <div className="absolute right-[-50%]">
          <TeamLogo
            teamId={awayTeam.teamId}
            teamName={awayTeam.teamName}
            width={360}
            height={360}
            mode="D"
          />
        </div>
      </div>
      <div className="relative h-[360px] bg-black p-10 container mx-auto max-w-[920px] z-10">
        <div className="hero-layer hero-overlay" />
        <img
          src="https://cdn.nba.com/next/arenas/arena.jpg"
          className="hero-layer"
          alt="arena"
        />
        <div className=" relative z-20 grid gap-2">
          <p className="text-white text-sm">
            {convertDateToEt({ date: gameTime })}
          </p>
          <p className="text-white font-bold text-5xl uppercase">{`${awayTeam.teamName} @ ${homeTeam.teamName}`}</p>
          <p className="text-white">{`${arena?.arenaName}, ${arena?.arenaCity}, ${arena?.arenaState}`}</p>
        </div>
      </div>
      <div className="flex-grow opacity-20 min-w-[240px] relative grayscale hidden md:block">
        <div className="absolute left-[-50%]">
          <TeamLogo
            teamId={homeTeam.teamId}
            teamName={homeTeam.teamName}
            width={480}
            height={480}
            mode="D"
          />
        </div>
      </div>
    </div>
  );
};

export default GameInfo;
