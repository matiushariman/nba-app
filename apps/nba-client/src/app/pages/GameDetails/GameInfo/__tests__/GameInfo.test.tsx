import GameInfo from '..';
import { render } from '../../../../utils/testUtils';

jest.mock('../../../hooks/useGetBoxscore', () => ({
  selectArena: jest.fn().mockReturnValueOnce({
    arenaName: 'TD Garden',
    arenaCity: 'Boston',
    arenaState: 'MA',
  }),
  selectGameTime: jest.fn().mockReturnValueOnce('2022-12-10T22:00:00Z'),
  selectGameScore: jest.fn().mockReturnValueOnce({
    awayTeam: {
      teamId: '1',
      teamName: 'Lakers',
      score: '92',
    },
    homeTeam: {
      teamId: '2',
      teamName: 'Celtics',
      score: '131',
    },
    gameStatusText: 'FINAL',
  }),
}));

describe('pages/GameDetails/GameInfo', () => {
  it('should render GameInfo that matches snapshot', () => {
    const { baseElement } = render(<GameInfo />);

    expect(baseElement).toMatchSnapshot();
  });
});
