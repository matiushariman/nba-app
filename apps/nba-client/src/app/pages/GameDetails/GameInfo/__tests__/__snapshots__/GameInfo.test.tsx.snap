// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/GameInfo should render GameInfo that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="bg-black flex"
    >
      <div
        class="flex-grow opacity-20 min-w-[240px] relative grayscale hidden md:block"
      >
        <div
          class="absolute right-[-50%]"
        >
          <img
            alt="Lakers logo"
            height="360"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/1/global/D/logo.svg"
            width="360"
          />
        </div>
      </div>
      <div
        class="relative h-[360px] bg-black p-10 container mx-auto max-w-[920px] z-10"
      >
        <div
          class="hero-layer hero-overlay"
        />
        <img
          alt="arena"
          class="hero-layer"
          src="https://cdn.nba.com/next/arenas/arena.jpg"
        />
        <div
          class=" relative z-20 grid gap-2"
        >
          <p
            class="text-white text-sm"
          >
            December 10, 2022 5:00 PM
          </p>
          <p
            class="text-white font-bold text-5xl uppercase"
          >
            Lakers @ Celtics
          </p>
          <p
            class="text-white"
          >
            TD Garden, Boston, MA
          </p>
        </div>
      </div>
      <div
        class="flex-grow opacity-20 min-w-[240px] relative grayscale hidden md:block"
      >
        <div
          class="absolute left-[-50%]"
        >
          <img
            alt="Celtics logo"
            height="480"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/2/global/D/logo.svg"
            width="480"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
